ext/opentelemetry-ext-botocore/src/opentelemetry/ext/botocore/__init__.py

```diff
@@ -0,0 +1,208 @@
+        # instead?
+        # pylint: disable=attribute-defined-outside-init
+        self._tracer = get_tracer(
+            __name__, __version__, kwargs.get("tracer_provider")
+        )
+
+        wrap_function_wrapper(
+            "botocore.client",
+            "BaseClient._make_api_call",
+            self._patched_api_call,
+        )
+
+    def _uninstrument(self, **kwargs):
+        unwrap(BaseClient, "_make_api_call")
+
+    def _patched_api_call(self, original_func, instance, args, kwargs):
+
+        endpoint_name = deep_getattr(instance, "_endpoint._endpoint_prefix")
+
+        with self._tracer.start_as_current_span(
+            "{}.command".format(endpoint_name), kind=SpanKind.CONSUMER,
+        ) as span:
+
+            operation = None
+            if args:
+                operation = args[0]
+                span.resource = "%s.%s" % (endpoint_name, operation.lower())
+
+            else:
+                span.resource = endpoint_name
+
+            add_span_arg_tags(
+                span, endpoint_name, args, ARGS_NAME, TRACED_ARGS
+            )
+
+            region_name = deep_getattr(instance, "meta.region_name")
+
+            meta = {
+                "aws.agent": "botocore",
+                "aws.operation": operation,
+                "aws.region": region_name,
+            }
+            for key, value in meta.items():
+                span.set_attribute(key, value)
+
+            result = original_func(*args, **kwargs)
+
+            span.set_attribute(
+                "http.status_code",
+                result["ResponseMetadata"]["HTTPStatusCode"],
+            )
+            span.set_attribute(
+                "retry_attempts", result["ResponseMetadata"]["RetryAttempts"],
+            )
+
+            return result
+
+
+def unwrap(obj, attr):
+    function = getattr(obj, attr, None)
+    if (
+        function
+        and isinstance(function, ObjectProxy)
+        and hasattr(function, "__wrapped__")
+    ):
+        setattr(obj, attr, function.__wrapped__)
+
+
+def add_span_arg_tags(span, endpoint_name, args, args_names, args_traced):
+    def truncate_arg_value(value, max_len=1024):
+        """Truncate values which are bytes and greater than `max_len`.
+        Useful for parameters like "Body" in `put_object` operations.
+        """
+        if isinstance(value, bytes) and len(value) > max_len:
+            return b"..."
+
+        return value
+
+    if endpoint_name not in ["kms", "sts"]:
+        tags = dict(
+            (name, value)
+            for (name, value) in zip(args_names, args)
+            if name in args_traced
+        )
+        tags = flatten_dict(tags)
+        for key, value in {
+            k: truncate_arg_value(v)
+            for k, v in tags.items()
+            if k not in {"s3": ["params.Body"]}.get(endpoint_name, [])
+        }.items():
+            span.set_attribute(key, value)
+
+
+def flatten_dict(dict_, sep=".", prefix=""):
```

Comment:
also shared with the boto instrumentation.

Reply:
Same as above.